LandSandBoat/server
# [Event] Mog Bonanza

**_I affirm:_**

## What does this pull request do?
Implements the Mog Bonanza event


## Steps to test these changes
Purchase Period
* Purchase a Pearl (Moogle Chooses number)
* Purchase a Pearl (User-selected number)
* Attempt to buy pearl without enough gil
* Attempt to buy pearl when having the maximum pearls
* View prizes

Collection Period
* View prizes for ranks
* Trade an invalid item
* Trade a rank 1 pearl
* Trade a rank 2 pearl
* Trade a rank 3 pearl
* Do not select a prize
* Select a gil reward
* Select an item reward
* Select an item reward with full inventory

Off:
* Ensure no interactions occur
